src/languages/swift.js

```diff
@@ -96,7 +84,52 @@ export default function(hljs) {
         { begin: /\b0b([01]_*)+\b/ },
       ]
   };
-  SUBST.contains = [NUMBER];
+
+  // https://docs.swift.org/swift-book/ReferenceManual/LexicalStructure.html#grammar_string-literal
+  var ESCAPED_CHARACTER = (rawDelimiter = "") => ({
+    className: 'subst',
+    variants: [
+      { begin: concat(/\\/, rawDelimiter, /[0\\tnr"']/) },
+      { begin: concat(/\\/, rawDelimiter, /u\{[0-9a-fA-F]{1,8}\}/) }
+    ]
+  });
+  var ESCAPED_NEWLINE = (rawDelimiter = "") => ({
+    className: 'subst',
+    begin: concat(/\\/, rawDelimiter, /[\t ]*(?:[\r\n]|\r\n)/)
```

Comment:
Much easier to read.

Is this variation and nuance in ordering necessary though? Is the order truly sometimes \n\r?  Or could this be simplified to `\r?\n` as is more common?

Reply:
That's just what the official grammar says:

```
line-break → U+000A
line-break → U+000D
line-break → U+000D followed by U+000A
```

So `\r?\n` is not correct, as a single `\r` is allowed too.